radixdlt/src/main/java/com/radixdlt/consensus/VerifiedLedgerHeaderAndProof.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.radixdlt.consensus.bft.View;
+import com.radixdlt.crypto.Hash;
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.DsonOutput.Output;
+import com.radixdlt.serialization.SerializerConstants;
+import com.radixdlt.serialization.SerializerDummy;
+import com.radixdlt.serialization.SerializerId2;
+import java.util.Objects;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Ledger header with proof
+ */
+@Immutable
+@SerializerId2("ledger.verified_ledger_header_and_proof")
+public final class VerifiedLedgerHeaderAndProof implements Comparable<VerifiedLedgerHeaderAndProof> {
```

Comment:
As discussed, the equals(...) method and comparable interface are inconsistent.

Would be better, perhaps, if this class has had a static method `Comparator<VerifiedLedgerHeaderAndProof> orderedByEpochAndStateVersion()`